pkg/agent/datamodel/types.go

```diff
@@ -14,38 +14,167 @@ import (
+type PublicKey struct {
+	KeyData string `json:"keyData"`
+}
+
+// KeyVaultCertificate specifies a certificate to install
+// On Linux, the certificate file is placed under the /var/lib/waagent directory
+// with the file name <UppercaseThumbprint>.crt for the X509 certificate file
+// and <UppercaseThumbprint>.prv for the private key. Both of these files are .pem formatted.
+// On windows the certificate will be saved in the specified store.
+type KeyVaultCertificate struct {
+	CertificateURL   string `json:"certificateUrl,omitempty"`
+	CertificateStore string `json:"certificateStore,omitempty"`
+}
+
+// KeyVaultID specifies a key vault
+type KeyVaultID struct {
+	ID string `json:"id,omitempty"`
+}
+
+// KeyVaultSecrets specifies certificates to install on the pool
+// of machines from a given key vault
+// the key vault specified must have been granted read permissions to CRP
+type KeyVaultSecrets struct {
+	SourceVault       *KeyVaultID           `json:"sourceVault,omitempty"`
+	VaultCertificates []KeyVaultCertificate `json:"vaultCertificates,omitempty"`
+}
+
+// ImageReference represents a reference to an Image resource in Azure.
+type ImageReference struct {
+	Name           string `json:"name,omitempty"`
+	ResourceGroup  string `json:"resourceGroup,omitempty"`
+	SubscriptionID string `json:"subscriptionId,omitempty"`
+	Gallery        string `json:"gallery,omitempty"`
+	Version        string `json:"version,omitempty"`
+}
+
+// VMDiagnostics contains settings to on/off boot diagnostics collection
+// in RD Host
+type VMDiagnostics struct {
+	Enabled bool `json:"enabled"`
+
+	// Specifies storage account Uri where Boot Diagnostics (CRP &
+	// VMSS BootDiagostics) and VM Diagnostics logs (using Linux
+	// Diagnostics Extension) will be stored. Uri will be of standard
+	// blob domain. i.e. https://storageaccount.blob.core.windows.net/
+	// This field is readonly as ACS RP will create a storage account
+	// for the customer.
+	StorageURL *neturl.URL `json:"storageUrl"`
+}
+
+// OSType represents OS types of agents
+type OSType string
+
+// the OSTypes supported by vlabs
+const (
+	Windows OSType = "Windows"
+	Linux   OSType = "Linux"
+)
+
+// Distro represents Linux distro to use for Linux VMs
+type Distro string
+
+// Distro string consts
+const (
+	Ubuntu               Distro = "ubuntu"
+	Ubuntu1804           Distro = "ubuntu-18.04"
+	Ubuntu1804Gen2       Distro = "ubuntu-18.04-gen2"
+	RHEL                 Distro = "rhel"
+	CoreOS               Distro = "coreos"
+	AKS1604Deprecated    Distro = "aks"               // deprecated AKS 16.04 distro. Equivalent to aks-ubuntu-16.04.
+	AKS1804Deprecated    Distro = "aks-1804"          // deprecated AKS 18.04 distro. Equivalent to aks-ubuntu-18.04.
+	AKSDockerEngine      Distro = "aks-docker-engine" // deprecated docker-engine distro.
+	AKSUbuntu1604        Distro = "aks-ubuntu-16.04"
+	AKSUbuntu1804        Distro = "aks-ubuntu-18.04"
+	ACC1604              Distro = "acc-16.04"
```

Comment:
we don't need this too

Reply:
Removed in new PR: https://github.com/Azure/AgentBaker/pull/240